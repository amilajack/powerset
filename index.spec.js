const power = require('.');

const a = [1, 2, 3, 4];

describe('basic tests', () => {
  it('test 1', () => {
    const b = power([1, 2, 3, 4, 5], { filter: { none: [1, 4, 5] } });
    expect(b).toMatchSnapshot();
  });

  it('test 2', () => {
    const options = { limit: -1 };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 3', () => {
    const options = { limit: 0 };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 4', () => {
    const options = { limit: 1 };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 5', () => {
    const options = { limit: [0, 1] };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 6', () => {
    const options = { limit: 2 };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 7', () => {
    const options = { limit: [1, 5] };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 8', () => {
    const options = { limit: [-1, -2] };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 9', () => {
    const options = { limit: [a.length, -2] };
    const b = power(a, options);
    expect(b).toMatchSnapshot();
  });

  it('test 10', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }];
    const options = { sort: 'desc' };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });
  it('test 11', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }];
    const options = { key: 'id', empty: true };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });

  it('test 12', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }, { id___: 4 }];
    const options = { sort: 'desc', key: 'id' };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });

  it('test 13', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }, { id___: 4 }];
    const options = { sort: 'desc', key: 'id', keySafe: true };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });

  it('test 14', () => {
    const d = [1, 2, 3, 4, 5, 6];
    const options = { filter: 6 };
    const b = power(d, options);
    expect(b).toMatchSnapshot();
  });

  it('test 15', () => {
    const d = [1, 2, 3, 4, 5, 6];
    const options = { filter: [1, 2] };
    const b = power(d, options);
    expect(b).toMatchSnapshot();
  });

  it('test 16', () => {
    const d = [1, 2, 3, 4, 5, 6];
    const options = { filter: [1, 2, 5, 6] };
    const b = power(d, options);
    expect(b).toMatchSnapshot();
  });

  it('test 17', () => {
    const d = [1, 2, 3, 4, 5, 6];
    const options = { filter: { any: [1, 2] } };
    const b = power(d, options);
    expect(b).toMatchSnapshot();
  });

  it('test 18', () => {
    const d = [1, 2, 3, 4, 5, 6];
    const options = { filter: { none: [1, 2] } };
    const b = power(d, options);
    expect(b).toMatchSnapshot();
  });

  it('test 19', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }];
    const options = { filter: { any: [1, 2] }, filterKey: 'id' };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });

  it('test 20', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }];
    const options = { filter: { all: [1, 2] }, filterKey: 'id' };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });

  it('test 21', () => {
    const c = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }];
    const options = { filter: { none: [1, 2] }, filterKey: 'id' };
    const b = power(c, options);
    expect(b).toMatchSnapshot();
  });
});
