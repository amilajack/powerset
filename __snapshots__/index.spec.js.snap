// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic tests test 1 1`] = `
Array [
  Array [
    2,
    3,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
]
`;

exports[`basic tests test 2 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    2,
    3,
    4,
  ],
]
`;

exports[`basic tests test 3 1`] = `Array []`;

exports[`basic tests test 4 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
  Array [
    4,
  ],
]
`;

exports[`basic tests test 5 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
  Array [
    4,
  ],
]
`;

exports[`basic tests test 6 1`] = `
Array [
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
  ],
  Array [
    3,
    4,
  ],
]
`;

exports[`basic tests test 7 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
  ],
  Array [
    1,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
  ],
  Array [
    2,
  ],
  Array [
    3,
    4,
  ],
  Array [
    3,
  ],
  Array [
    4,
  ],
]
`;

exports[`basic tests test 8 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
  ],
  Array [
    3,
    4,
  ],
]
`;

exports[`basic tests test 9 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
  ],
  Array [
    3,
    4,
  ],
]
`;

exports[`basic tests test 10 1`] = `
Array [
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
  ],
  Array [
    Object {
      "id": 3,
    },
  ],
]
`;

exports[`basic tests test 11 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
  Array [],
]
`;

exports[`basic tests test 12 1`] = `
Array [
  Array [
    1,
    2,
    3,
    undefined,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    undefined,
  ],
  Array [
    1,
    3,
    undefined,
  ],
  Array [
    2,
    3,
    undefined,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    undefined,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    undefined,
  ],
  Array [
    3,
    undefined,
  ],
  Array [
    1,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
  Array [
    undefined,
  ],
]
`;

exports[`basic tests test 13 1`] = `
Array [
  Array [
    1,
    2,
    3,
    undefined,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    undefined,
  ],
  Array [
    1,
    3,
    undefined,
  ],
  Array [
    2,
    3,
    undefined,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    undefined,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    undefined,
  ],
  Array [
    3,
    undefined,
  ],
  Array [
    1,
  ],
  Array [
    2,
  ],
  Array [
    3,
  ],
  Array [
    undefined,
  ],
]
`;

exports[`basic tests test 14 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
    6,
  ],
  Array [
    1,
    2,
    3,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    6,
  ],
  Array [
    1,
    2,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    4,
    6,
  ],
  Array [
    1,
    2,
    5,
    6,
  ],
  Array [
    1,
    2,
    6,
  ],
  Array [
    1,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    3,
    4,
    6,
  ],
  Array [
    1,
    3,
    5,
    6,
  ],
  Array [
    1,
    3,
    6,
  ],
  Array [
    1,
    4,
    5,
    6,
  ],
  Array [
    1,
    4,
    6,
  ],
  Array [
    1,
    5,
    6,
  ],
  Array [
    1,
    6,
  ],
  Array [
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    2,
    3,
    4,
    6,
  ],
  Array [
    2,
    3,
    5,
    6,
  ],
  Array [
    2,
    3,
    6,
  ],
  Array [
    2,
    4,
    5,
    6,
  ],
  Array [
    2,
    4,
    6,
  ],
  Array [
    2,
    5,
    6,
  ],
  Array [
    2,
    6,
  ],
  Array [
    3,
    4,
    5,
    6,
  ],
  Array [
    3,
    4,
    6,
  ],
  Array [
    3,
    5,
    6,
  ],
  Array [
    3,
    6,
  ],
  Array [
    4,
    5,
    6,
  ],
  Array [
    4,
    6,
  ],
  Array [
    5,
    6,
  ],
  Array [
    6,
  ],
]
`;

exports[`basic tests test 15 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
    5,
  ],
  Array [
    1,
    2,
    3,
    4,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    1,
    2,
    3,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    5,
  ],
  Array [
    1,
    2,
    3,
    6,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    4,
    5,
  ],
  Array [
    1,
    2,
    4,
    6,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
    5,
    6,
  ],
  Array [
    1,
    2,
    5,
  ],
  Array [
    1,
    2,
    6,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    3,
    4,
    5,
  ],
  Array [
    1,
    3,
    4,
    6,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
    5,
    6,
  ],
  Array [
    1,
    3,
    5,
  ],
  Array [
    1,
    3,
    6,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
    5,
    6,
  ],
  Array [
    1,
    4,
    5,
  ],
  Array [
    1,
    4,
    6,
  ],
  Array [
    1,
    4,
  ],
  Array [
    1,
    5,
    6,
  ],
  Array [
    1,
    5,
  ],
  Array [
    1,
    6,
  ],
  Array [
    1,
  ],
  Array [
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    2,
    3,
    4,
    5,
  ],
  Array [
    2,
    3,
    4,
    6,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
    5,
    6,
  ],
  Array [
    2,
    3,
    5,
  ],
  Array [
    2,
    3,
    6,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
    5,
    6,
  ],
  Array [
    2,
    4,
    5,
  ],
  Array [
    2,
    4,
    6,
  ],
  Array [
    2,
    4,
  ],
  Array [
    2,
    5,
    6,
  ],
  Array [
    2,
    5,
  ],
  Array [
    2,
    6,
  ],
  Array [
    2,
  ],
]
`;

exports[`basic tests test 16 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
    5,
  ],
  Array [
    1,
    2,
    3,
    4,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    1,
    2,
    3,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    5,
  ],
  Array [
    1,
    2,
    3,
    6,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    4,
    5,
  ],
  Array [
    1,
    2,
    4,
    6,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
    5,
    6,
  ],
  Array [
    1,
    2,
    5,
  ],
  Array [
    1,
    2,
    6,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    3,
    4,
    5,
  ],
  Array [
    1,
    3,
    4,
    6,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
    5,
    6,
  ],
  Array [
    1,
    3,
    5,
  ],
  Array [
    1,
    3,
    6,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
    5,
    6,
  ],
  Array [
    1,
    4,
    5,
  ],
  Array [
    1,
    4,
    6,
  ],
  Array [
    1,
    4,
  ],
  Array [
    1,
    5,
    6,
  ],
  Array [
    1,
    5,
  ],
  Array [
    1,
    6,
  ],
  Array [
    1,
  ],
  Array [
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    2,
    3,
    4,
    5,
  ],
  Array [
    2,
    3,
    4,
    6,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
    5,
    6,
  ],
  Array [
    2,
    3,
    5,
  ],
  Array [
    2,
    3,
    6,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
    5,
    6,
  ],
  Array [
    2,
    4,
    5,
  ],
  Array [
    2,
    4,
    6,
  ],
  Array [
    2,
    4,
  ],
  Array [
    2,
    5,
    6,
  ],
  Array [
    2,
    5,
  ],
  Array [
    2,
    6,
  ],
  Array [
    2,
  ],
  Array [
    3,
    4,
    5,
    6,
  ],
  Array [
    3,
    4,
    5,
  ],
  Array [
    3,
    4,
    6,
  ],
  Array [
    3,
    5,
    6,
  ],
  Array [
    3,
    5,
  ],
  Array [
    3,
    6,
  ],
  Array [
    4,
    5,
    6,
  ],
  Array [
    4,
    5,
  ],
  Array [
    4,
    6,
  ],
  Array [
    5,
    6,
  ],
  Array [
    5,
  ],
  Array [
    6,
  ],
]
`;

exports[`basic tests test 17 1`] = `
Array [
  Array [
    1,
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
    5,
  ],
  Array [
    1,
    2,
    3,
    4,
    6,
  ],
  Array [
    1,
    2,
    3,
    4,
  ],
  Array [
    1,
    2,
    3,
    5,
    6,
  ],
  Array [
    1,
    2,
    3,
    5,
  ],
  Array [
    1,
    2,
    3,
    6,
  ],
  Array [
    1,
    2,
    3,
  ],
  Array [
    1,
    2,
    4,
    5,
    6,
  ],
  Array [
    1,
    2,
    4,
    5,
  ],
  Array [
    1,
    2,
    4,
    6,
  ],
  Array [
    1,
    2,
    4,
  ],
  Array [
    1,
    2,
    5,
    6,
  ],
  Array [
    1,
    2,
    5,
  ],
  Array [
    1,
    2,
    6,
  ],
  Array [
    1,
    2,
  ],
  Array [
    1,
    3,
    4,
    5,
    6,
  ],
  Array [
    1,
    3,
    4,
    5,
  ],
  Array [
    1,
    3,
    4,
    6,
  ],
  Array [
    1,
    3,
    4,
  ],
  Array [
    1,
    3,
    5,
    6,
  ],
  Array [
    1,
    3,
    5,
  ],
  Array [
    1,
    3,
    6,
  ],
  Array [
    1,
    3,
  ],
  Array [
    1,
    4,
    5,
    6,
  ],
  Array [
    1,
    4,
    5,
  ],
  Array [
    1,
    4,
    6,
  ],
  Array [
    1,
    4,
  ],
  Array [
    1,
    5,
    6,
  ],
  Array [
    1,
    5,
  ],
  Array [
    1,
    6,
  ],
  Array [
    1,
  ],
  Array [
    2,
    3,
    4,
    5,
    6,
  ],
  Array [
    2,
    3,
    4,
    5,
  ],
  Array [
    2,
    3,
    4,
    6,
  ],
  Array [
    2,
    3,
    4,
  ],
  Array [
    2,
    3,
    5,
    6,
  ],
  Array [
    2,
    3,
    5,
  ],
  Array [
    2,
    3,
    6,
  ],
  Array [
    2,
    3,
  ],
  Array [
    2,
    4,
    5,
    6,
  ],
  Array [
    2,
    4,
    5,
  ],
  Array [
    2,
    4,
    6,
  ],
  Array [
    2,
    4,
  ],
  Array [
    2,
    5,
    6,
  ],
  Array [
    2,
    5,
  ],
  Array [
    2,
    6,
  ],
  Array [
    2,
  ],
]
`;

exports[`basic tests test 18 1`] = `
Array [
  Array [
    3,
    4,
    5,
    6,
  ],
  Array [
    3,
    4,
    5,
  ],
  Array [
    3,
    4,
    6,
  ],
  Array [
    3,
    4,
  ],
  Array [
    3,
    5,
    6,
  ],
  Array [
    3,
    5,
  ],
  Array [
    3,
    6,
  ],
  Array [
    3,
  ],
  Array [
    4,
    5,
    6,
  ],
  Array [
    4,
    5,
  ],
  Array [
    4,
    6,
  ],
  Array [
    4,
  ],
  Array [
    5,
    6,
  ],
  Array [
    5,
  ],
  Array [
    6,
  ],
]
`;

exports[`basic tests test 19 1`] = `
Array [
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 2,
    },
  ],
]
`;

exports[`basic tests test 20 1`] = `
Array [
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
  ],
]
`;

exports[`basic tests test 21 1`] = `
Array [
  Array [
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
  ],
  Array [
    Object {
      "id": 3,
    },
  ],
  Array [
    Object {
      "id": 4,
    },
  ],
]
`;
